spec/requests/irs_attempts_api_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'IRS attempts API' do
+  before do
+    IrsAttemptsApi::RedisClient.clear_attempts!
+    events_to_acknowledge
+    events_to_render
+  end
+
+  let(:events_to_acknowledge) do
```

Comment:
if we switch the `let` to `let!(` we can let Rspec eager load these and simplify the before block:

```suggestion
  end

  let!(:events_to_acknowledge) do
```

Reply:
Okay, so this gets tricky because of the `.clear_attempts!` call in the before action. That runs _after_ the `let!` block so the events get created and then immediately destroyed 😭